src/auth/basic/LDAP/basic_ldap_auth.cc

```diff
@@ -602,7 +610,7 @@ main(int argc, char **argv)
                 ld = NULL;
                 goto recover;
             }
-            printf("ERR %s\n", ldap_err2string(squid_ldap_errno(ld)));
+            printf("%s %s\n", (!bherror || LDAP_SECURITY_ERROR(squid_ldap_errno(ld))) ? "ERR" : "BH", ldap_err2string(squid_ldap_errno(ld)));
```

Comment:
Please do this as a conversion from printf() statements in this while-loop to the provided SEND_*() macros in helper/protocol_defines.h which are the official helper API interface for Squid-3+.

This particular line change would then look like this:
```suggestion
            const auto e = squid_ldap_errno(ld);
            if (LDAP_SECURITY_ERROR(e)) {
                SEND_ERR(HLP_MSG(ldap_err2string(e)));
            } else {
                SEND_BH(HLP_MSG(ldap_err2string(e)));
            }
```